### PR TITLE
Mass deployments script/workflow

### PR DESCRIPTION
**Pull request by mohamedamer453**
### Description

Adding a mass deployment workflow and script to randomly create deployments on different farms and nodes.

### Changes

- added batch_vms script.
- added mass_deployments workflow.

### Related Issues

#1282 

### Checklist

- [ ] Tests included
- [x] Build pass
- [ ] Documentation
- [x] Code format and docstrings
- [ ] Screenshots/Video attached (needed for UI changes)

### CODE REVIEW

File: .github/workflows/mass_deployments.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    - cron: "0 1 * * *"
+  workflow_dispatch:
+
+jobs:
+  mass-deployments:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+    env:
+      NETWORK: main
```
Comment (0oM4R) [inline .github/workflows/mass_deployments.yml:14]: is this script will run only on main net? if no, we should handle this line

Comment (zaelgohary) [reply .github/workflows/mass_deployments.yml:14]: As the [issue](https://github.com/threefoldtech/tfgrid-sdk-ts/issues/1282) states, it'll run only on mainnet.

**Review by 0oM4R — CHANGES_REQUESTED** [review]
I tried to test the workflow : https://github.com/threefoldtech/tfgrid-sdk-ts/actions/runs/6683847114/job/18160498772

and it fails We think this is because it checks out v2.1.1 and this version doesn't have batch_vms.ts script

File: .github/workflows/mass_deployments.yml [hunk 2]
```diff
@@ -1,8 +1,7 @@
 name: Mass Deployments
 
 on:
-  schedule:
-    - cron: "0 1 * * *"
+  push
```
Comment (xmonader) [inline .github/workflows/mass_deployments.yml]: Why? this shouldn't run on every push!

Comment (mohamedamer453) [reply .github/workflows/mass_deployments.yml]: Just updated it to on push to test the workflow and ensure that it's running, once the following run https://github.com/threefoldtech/tfgrid-sdk-ts/actions/runs/6684027673/job/18160843043 is done, i'll revert this change.

**Review by AhmedHanafy725 — CHANGES_REQUESTED** [review]

File: .github/workflows/mass_deployments.yml (hunk 2 above)
Comment (mohamedamer453) [reply .github/workflows/mass_deployments.yml]: updated in https://github.com/threefoldtech/tfgrid-sdk-ts/pull/1289/commits/c94b27b6be84381880bf490155ba97fdad89cbd3

File: packages/grid_client/scripts/batch_vms.ts [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    const disk1 = new DiskModel();
+    disk1.name = "d" + generateString(5);
+    disk1.size = diskSize;
+    disk1.mountpoint = "/newDisk1";
+
+    //Farm Selection
+    const farms = await grid3.capacity.filterFarms({
+      nodeMRU: mru / 1024,
+      nodeSRU: diskSize + rootFs,
+      publicIp: publicIp,
```
[the remark this answers is not in the document]
Comment (mohamedamer453) [reply packages/grid_client/scripts/batch_vms.ts]: both the `randomize` parameter and `available_for` were added in https://github.com/threefoldtech/tfgrid-sdk-ts/pull/1289/commits/f1f096915712581bf39f87792e41c2bcae0bd490

File: packages/grid_client/scripts/batch_vms.ts
[the remark this answers is not in the document]
Comment (mohamedamer453) [reply packages/grid_client/scripts/batch_vms.ts:1]: the script was renamed in https://github.com/threefoldtech/tfgrid-sdk-ts/pull/1289/commits/f1f096915712581bf39f87792e41c2bcae0bd490

File: packages/grid_client/scripts/batch_vms.ts [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    } as FarmFilterOptions);
+
+    //Node Selection
+    const nodes = await grid3.capacity.filterNodes({
+      cru: cru,
+      mru: mru / 1024,
+      sru: rootFs + diskSize,
+      availableFor: await grid3.twins.get_my_twin_id(),
+      farmId: +randomChoice(farms).farmId,
```
[the remark this answers is not in the document]
Comment (mohamedamer453) [reply packages/grid_client/scripts/batch_vms.ts]: conditions were added to check for the availability of the returned farms and nodes in https://github.com/threefoldtech/tfgrid-sdk-ts/pull/1289/commits/f1f096915712581bf39f87792e41c2bcae0bd490

**Comment by mohamedamer453** [comment]
The workflow was tested and successfully passed in https://github.com/threefoldtech/tfgrid-sdk-ts/actions/runs/6684027673

**Review by 0oM4R — APPROVED** [review]
Good job ya   :100:

**Review by AhmedHanafy725 — APPROVED** [review]

**Pull request merged**